pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -179,6 +180,11 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterf
 		// Get the current resource list from discovery.
 		newResources := GetDeletableResources(discoveryClient)
 
+		// Filter resources that have been uninstalled and reinstalled on the cluster since the last sync.
+		// The following sync will recognize them as new resources and they will be restarted.
+		// If we do not filter them out, the informer that was previously stopped will not get restarted.
+		gc.dependencyGraphBuilder.filterStoppedResources(newResources)
```

Comment:
this mutates both the input variable and `dependencyGraphBuilder` state in a way that I find very hard to reason about. (it also happens before we've done sanity checks on `newResources` to make sure we even got results and that they differ from our last check)

I expected no-longer-existing resources to be handled in the guts of `syncMonitors` (where we're currently calculating `toRemove` already based on delta from the last set of resources we started monitors for)